module/zfs/vdev_disk.c

```diff
@@ -570,7 +570,7 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio, caddr_t kbuf_ptr,
 	/* Extra reference to protect dio_request during vdev_submit_bio */
 	vdev_disk_dio_get(dr);
 	if (zio)
-		zio->io_delay = jiffies_64;
+		zio->io_disk_delay = gethrtime();
```

Comment:
`io_disk_delay` is only use for a single error message in `vdev_disk.c`:

  ```
struct zio {
...
        hrtime_t        io_delay;       /* Device access time (disk or */
                                        /* file). */
        hrtime_t        io_disk_delay;  /* Same as io_delay, but for disks */
                                        /* only.  Used for printing the */
                                        /* delay in error messages in */
                                        /* vdev_disk.c */
```
`io_delay` is what you're looking for, and it is calculated at a higher level (in `zio_vdev_io_start()`/`zio_vdev_io_done()`).  At that level, it does include file-based vdevs.